sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -111,6 +113,63 @@ object RapidsPluginUtils extends Logging {
     }
   }
 
+  private def detectMultipleJar(propName: String, jarName: String, conf: RapidsConf): Unit = {
+    val classloader = ShimLoader.getShimClassLoader()
+    val possibleRapidsJarURLs = classloader.getResources(propName).asScala.toSet.toSeq.filter {
+      url => {
+        val urlPath = url.toString
+        // filter out submodule jars and files stored under subdirs of '!/'
+        !urlPath.contains("rapids-4-spark-") && urlPath.contains("!/" + propName)
+      }
+    }
```

Comment:
Not too easy to follow but I get it now. In the comment can you add an example of the submodule vs main dist artifact you are looking for. 

However, now it looks like we NEVER will never find out where we put submodule jars for tests by accident?

Reply:
> However, now it looks like we NEVER will never find out where we put submodule jars for tests by accident?

How about adding another config mode like RECURISIVE to detect all jars on classpath and print a warning log? Just need to skip this filter in that mode.

Reply:
What do you mean by RECURSIVE?

if RECURSIVE  means going down a directory level inside a jar file, I think we already established that we should only look at the root entries 
ending with `"!/" + propName`

if RECURSIVE means including the check to all sub-module artitfact id  prefixed with `rapids-4-spark-` I realize that we do not have a good story now that change of the version info file name to contain the artifact Id is not part of the PR. Let us defer this to the followup issue you were going to file.

Reply:
> if RECURSIVE means including the check to all sub-module artitfact id prefixed with rapids-4-spark- I realize that we do not have a good story now that change of the version info file name to contain the artifact Id is not part of the PR. Let us defer this to the followup issue you were going to file.

Yes, I mean this way. I will add it to the follow-up issue too.